chainercv/extensions/detection/detection_vis_report.py

```diff
@@ -27,179 +22,56 @@ def _check_available():
                       '  $ pip install matplotlib\n')
 
 
-def _detection_vis_transform(xs):
-    return xs[0].astype(np.uint8), xs[1], xs[2]
-
-
 class DetectionVisReport(chainer.training.extension.Extension):
 
     """An extension that visualizes output of a detection model.
 
     This extension visualizes the predicted bounding boxes together with the
     ground truth bounding boxes.
-    The examples used in visualization are selected by ids included in
-    :obj:`indices`.
-    This extension wraps three steps of operations needed to visualize output
-    of the model. :obj:`dataset`,
-    :obj:`predict_func` and :obj:`vis_transform` are used at each step.
-
-    1. Getting an example.
-        :meth:`dataset.__getitem__` returns tuple of arrays that are used as
-        the arguments for :obj:`predict_func`.
-
-        .. code:: python
-
-            inputs = dataset[i]
-            img, bbox, label = inputs
-
-        :obj:`i` corresponds to an id included in :obj:`indices`.
-
-    2. Predicting the output.
-        Given the inputs, :meth:`predict_func` returns a prediction.
-        :meth:`predict_func` should accept the first element of the tuple
-        returned by the dataset, and returns a prediction.
-        A batch axis is added to the input image when fed into the function
-        as an argument. The prediction returned by the function can be with
-        or without batch axis. The code below illustrates this step.
-
-        .. code:: python
-
-            img = inputs[0]  # first element of the tuple
-            # pred_bbox can be either (1, R, 4) or (R, 4)
-            # pred_label can be either (1, R) or (R,)
-            pred_bbox, pred_label = predict_func(img[None])
-
-    3. Converting input arrays for visualization.
-        Given the inputs from :meth:`dataset.__getitem__`, a method
-        :meth:`vis_transform` should convert them into visualizable forms.
-        The values returned by :meth:`vis_transform` should be
-
-        .. code:: python
-
-            img, bbox, label = vis_transform(inputs)
-
-        :obj:`img` should be an image which is in CHW format, BGR,
-        :math:`[0, 255]` and :obj:`dtype==numpy.uint8`.
 
     The process can be illustrated in the following code.
 
     .. code:: python
 
-        img, bbox, label = dataset[i]
-        pred_bbox, pred_label = predict_func((img[None]))
-        pred_bbox = pred_bbox[0]  # (B, R, 4) -> (R, 4)
-        pred_label = pred_label[0]  # (B, R)  -> (R,)
-        vis_img, vis_bbox, vis_label = vis_transform(inputs)
+        batch = next(iterator)
+        # Convert batch -> imgs, gt_bboxes, gt_labels
+        pred_bboxes, pred_labels, pred_scores = target.predict(imgs)
         # Visualization code
-        # Uses (vis_img, vis_bbox, vis_label) as the ground truth output
-        # Uses (vis_img, pred_bbox, pred_label) as the predicted output
```

Comment:
Variables with single objects (e.g. `img, gt_bbox`) have not appeared previously in the doc.

Can you add something like

```
# Visualization code
for img, gt_bbox, gt_label, pred_bbox, pred_label, pred_score in ... :
    vis_bbox(img, gt_bbox, gt_label)
    vis_bbox(img, pred_bbox, pred_label, pred_score)
```